api/db/changelog/db.changelog-225-add-user-discovery-source.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+  <changeSet author="Peter-Lavigne" id="changelog-225-add-user-discovery-source">
+    <addColumn tableName="user">
+      <column name="user_general_discovery_source_other_text" type="varchar(255)"/>
+      <column name="user_partner_discovery_source_other_text" type="varchar(255)"/>
+    </addColumn>
+
+    <createTable tableName="user_general_discovery_source">
+      <column name="user_id" type="bigint">
+        <constraints
+                nullable="false"
+                foreignKeyName="fk_user_general_discovery_source_user"
+                references="user(user_id)"
+                deleteCascade="true"
+        />
+      </column>
+
+      <column name="source" type="ENUM('RESEARCH_ALL_OF_US_WEBSITE','SOCIAL_MEDIA','JOURNAL_OR_NEWS_ARTICLE','ACTIVITY_PRESENTATION_OR_EVENT','FRIENDS_OR_COLLEAGUES','OTHER_WEBSITE','OTHER')">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+
+    <createTable tableName="user_partner_discovery_source">
+      <column name="user_id" type="bigint">
+        <constraints
+                nullable="false"
+                foreignKeyName="fk_user_partner_discovery_source_user"
+                references="user(user_id)"
+                deleteCascade="true"
+        />
+      </column>
+      <column name="source" type="ENUM(
+        'ALL_OF_US_RESEARCH_PROGRAM_STAFF',
```

Comment:
totally optional. everything else is ordered except for `ALL_OF_US_RESEARCHER_ACADEMY_RTI_INTERNATIONAL`

Reply:
Ha, [Joel had the same comment](https://pmi-engteam.slack.com/archives/C02MWP2RN5P/p1692892734014169?thread_ts=1692892056.905769&cid=C02MWP2RN5P), which I only updated in the UI. I'll change the enum orderings to match.

Reply:
[Joel had the same comment](https://pmi-engteam.slack.com/archives/C02MWP2RN5P/p1692892734014169?thread_ts=1692892056.905769&cid=C02MWP2RN5P), but I only updated the UI. I'll change the enum orderings to match.

Reply:
Done 👍